### ISSUE TITLE
Error Code List

### ISSUE THREAD

**Issue by chaosisbliss** [MSEndpointMgr/ConfigMgr#253]
I remember there being a list of error codes on the old site that we could use to help identify issues. I cannot seem to find that list anymore. Since updating to 6.4.8 and using  verion 3.0.4 Invoke-CMApplyDriverPackage.ps1, I seem to be getting error code 1 on 90% of images. 

![image](https://user-images.githubusercontent.com/51373358/88400698-51399600-cd8e-11ea-8a2e-da6aebeb1e60.png)

**Comment by NickolajA** [comment]
Since version 3.x the error codes was removed and replaced by a single error 1. Check the real error as it should be outputted to the ApplyDriverPackage.log file.

**Comment by chaosisbliss** [comment]
Ah. Thanks for the quick response. I've looked at about 50 different ApplyDriverPackage logs as I have them dumped at the end of the task sequence.  They don't indicate the issue. Processing driver package with 4 detection methods: Drivers - Driver package name. Then goes to the next line,  Apply Driver package failed,  please refer to previous error or warning messages. Everything prior is successful. It detects the model, sku, target build.

**Comment by NickolajA** [comment]
https://github.com/MSEndpointMgr/ModernDriverManagement/blob/master/Invoke-CMApplyDriverPackage_Legacy.ps1

Please try version 3.0.6 (remember to rename the file and the remove the '_Legacy' part). It should contain, hopefully, a bit more logging information for where I believe the error to occur.

**Comment by chaosisbliss** [comment]
Great ! Thank you sir! I've got everything copied over and will keep you updated.

**Comment by chaosisbliss** [comment]
It is saying the invoke-cmapplydriverpackage.ps1 to the -File parameter does not exist. Provide the path to an existing .ps1 file argument to the -File parameter. Failed to run the action.  Error -196608.

**Comment by NickolajA** [comment]
I'd assume the same error is present in the ApplyDriverPackage.log, could you verify and paste a screenshot of it?

**Comment by chaosisbliss** [comment]
My fault on the above error code. I put and underscore instead of a hypen when changing the new script. We're back to the original error code 1 message. 

SMSTS.log
![image](https://user-images.githubusercontent.com/51373358/88414997-5f45e180-cda3-11ea-9da6-e1718ec06e2f.png)


ApplyDriverPackage.log
![image](https://user-images.githubusercontent.com/51373358/88414863-27d73500-cda3-11ea-895f-2464d69e95b5.png)

**Comment by vbateman** [comment]
I just noticed something, why are you running this as install software, when you could just use the action in OSD to run powershell. Could that be the problem.

**Comment by vbateman** [comment]
![powershell](https://user-images.githubusercontent.com/19505431/88416069-c1e7af00-cd9c-11ea-9f8f-9d6e2fb26b9b.png)

**Comment by chaosisbliss** [comment]
I have two different options to install drivers and firmware both of which are producing the same error.  I have one that is used during PXE to install drivers and firmware. Logs are showing the same errors. The second one is used Post PXE so our support reps can update machines while windows is installed. Logs are showing the same errors.

**Comment by NickolajA** [comment]
What you're doing is right, there's something about the package it's processing that the code doesn't like and bails out. I thought it was in the part where it constructs a custom object of all driver package properties that's combined with local data of the device used for the matching process in the next phase. It might be the later stage where it does some HP 'magic' that's required. Let me download that package and run it in debug mode and get back to you. In the meantime, could you please take a screenshot of the properties of the package from ConfigMgr?

**Comment by chaosisbliss** [comment]
Good deal. That makes me feel better. 

![image](https://user-images.githubusercontent.com/51373358/88420574-b00e0800-cdac-11ea-9ec6-f81d7f1a04f8.png)

**Comment by Noizcontroller** [comment]
@NickolajA I'm trying to setup MDM but I'm having the same issue when running the "Apply driver package" step. The task sequence fails and outputs "1". There is no useful information in the ApplyDrivePackage.log like chaosisbliss said. I've read in another issue (https://github.com/MSEndpointMgr/ConfigMgr/issues/243#issuecomment-664565089) that there is a version 4.x of the Powershell script but that the documentation has to be updated on the website. Any idea when this will happen?

**Comment by chaosisbliss** [comment]
Just making sure I didn't miss any updates to the invoke package. Should we revert back to the older version for the time being?

**Comment by chaosisbliss** [comment]
I had to revert back to the 2.2.7 invoke-applydriverpackage.ps1 file.

**Comment by Ar5had07** [comment]
I am having the problem.  I can't build any devices atm.

![image](https://user-images.githubusercontent.com/46709367/89529677-08013180-d7e5-11ea-8914-7b65aeb23ff6.png)
![image](https://user-images.githubusercontent.com/46709367/89529692-0fc0d600-d7e5-11ea-863c-0b670a4f4bfb.png)
![image](https://user-images.githubusercontent.com/46709367/89529771-2e26d180-d7e5-11ea-843c-fdac312776ca.png)

**Comment by NickolajA** [comment]
I may have found an issue where HP have not done their homework really, and put a space-char instead of a comma-space in the list of SKU's. The script has been updated to 3.0.7, but only available in the new repo:

https://github.com/MSEndpointMgr/ModernDriverManagement/blob/master/Invoke-CMApplyDriverPackage_Legacy.ps1

Not sure if this is the same issue that you're describing here though.

@Ar5had07: Your log shows that the script crashes on the 1040 G4 driver package, could you post the package properties of that?

**Comment by NickolajA** [comment]
I've found the issue causing this behavior and is working on a fix. Stay tuned.

**Comment by Ar5had07** [comment]
> I may have found an issue where HP have not done their homework really, and put a space-char instead of a comma-space in the list of SKU's. The script has been updated to 3.0.7, but only available in the new repo:
> 
> https://github.com/MSEndpointMgr/ModernDriverManagement/blob/master/Invoke-CMApplyDriverPackage_Legacy.ps1
> 
> Not sure if this is the same issue that you're describing here though.
> 
> @Ar5had07: Your log shows that the script crashes on the 1040 G4 driver package, could you post the package properties of that?

Here you go 

![image](https://user-images.githubusercontent.com/46709367/89631973-6ab40500-d899-11ea-8d73-4e1777be49b6.png)

**Comment by Ar5had07** [comment]
> I may have found an issue where HP have not done their homework really, and put a space-char instead of a comma-space in the list of SKU's. The script has been updated to 3.0.7, but only available in the new repo:
> 
> https://github.com/MSEndpointMgr/ModernDriverManagement/blob/master/Invoke-CMApplyDriverPackage_Legacy.ps1
> 

Just tried the Legacy.ps1 and same error

![image](https://user-images.githubusercontent.com/46709367/89632847-cf239400-d89a-11ea-9f19-4c6edee27c8f.png)

**Comment by NickolajA** [comment]
Yup, that's the exact error. I've spoken with Maurice too so that we verify it's not a change in how HP is exposing their data in the XML feed. Anyway, we now know what the issue is and you can expect a fix shortly.

**Comment by NickolajA** [comment]
Check out version 3.0.8 that was just committed, it shouldn't crash like this in that version.

**Comment by Ar5had07** [comment]
> Check out version 3.0.8 that was just committed, it shouldn't crash like this in that version.

Thank you, that's now working.  But the detection of models isn't.  It's not detecting HP 800 G2 Small Form Factor

![image](https://user-images.githubusercontent.com/46709367/89741443-4dde2400-da89-11ea-862d-30288e1e6a5e.png)

**Comment by Ar5had07** [comment]
> Check out version 3.0.8 that was just committed, it shouldn't crash like this in that version.

I renamed the package in ConfigMgr to force it to match, but failed somewhere else.

![image](https://user-images.githubusercontent.com/46709367/89741496-b4fbd880-da89-11ea-954e-d07cc45aed28.png)

**Comment by Ar5had07** [comment]
^ it crashed on 800 G3 because it didn't have a models in the description.  I deleted that package.  I ran the script again and this time, this error.

![image](https://user-images.githubusercontent.com/46709367/89741560-289de580-da8a-11ea-8efd-cc9fffe8a802.png)

**Comment by NickolajA** [comment]
Version 3.x of the script should have filtered out driver packages that didn't have any data in the Comments/Description field, could you verify it did? It's logged right before the processing starts.

Are you using the -TargetOSVersion "18363" parameter with input for 1909, so that it's able to detect?

**Comment by Ar5had07** [comment]
> -TargetOSVersion "18363"

Right, it's now working when I add the TargetOSVersion, but it's not filtering out packages that don't have Comments/Description.  (I've had to delete a package that dont have comments for it to work)
This is the beginning of the log
![image](https://user-images.githubusercontent.com/46709367/89741732-af06f700-da8b-11ea-9d91-7ab4f12e773d.png)

**Comment by tunnelofsludge** [comment]
I also seem to show that adding -TargetOSVersion "18363" to the command line, works. This may cause older driverpacks (1809/1903) to fail, but I haven't tested it.

**Comment by chaosisbliss** [comment]
Hi,

I just updated to the newest 4.06 script and updating my test task sequences to verify functionality. I seem to be hitting a snag. According to (Get-CIMInstance -ClassName MS_SystemInformation -NameSpace root\WMI).BaseBoardProduct, my returned value is 83D5 which is a HP EliteBook 755 G5. Using the -debuging option running the .\Invoke.ps1, which is amazing by the way, I get 0 returned packages. The driver package exists as well. I've deleted the old packages so the newly update 6.4.9 could download it again in case there were any questions. 
.\Invoke-CMApplyDriverPackage.ps1 -DebugMode -Endpoint "ServerName" -UserName "Username Here" -Password "PasswordHere" -Manufacturer "HP" -ComputerModel "HP EliteBook 755 G5" -SystemSKU "83D5"

OSApplyDriverPackage.log file generated from the debug.
![image](https://user-images.githubusercontent.com/51373358/97217608-8ee6fa00-1795-11eb-91d6-3b0d91241dfb.png)

Package in MECM. 
![image](https://user-images.githubusercontent.com/51373358/97217724-b0e07c80-1795-11eb-9b5a-4cc697239d7e.png)
